tests/unit/index_tests.py

```diff
@@ -584,5 +585,134 @@ def test_delete_disabled(self):
             'Deleting the \"special\" index is not allowed.'
         )
 
+@unittest.skipUnless(
+    os.environ.get('RUN_CLOUDANT_TESTS') is not None,
+    'Skipping Cloudant Search Index tests'
+    )
+class SearchIndexTests(UnitTestDbBase):
+    """
+    Search index unit tests
+    """
+
+    def setUp(self):
+        """
+        Set up test attributes
+        """
+        super(SearchIndexTests, self).setUp()
+        self.db_set_up()
+
+    def tearDown(self):
+        """
+        Reset test attributes
+        """
+        self.db_tear_down()
+        super(SearchIndexTests, self).tearDown()
+
+    def test_constructor(self):
+        """
+        Test instantiating a SearchIndex
+        """
+        ddoc = DesignDocument(self.db, 'ddoc001')
+        search = SearchIndex(
+            ddoc,
+            'searchindex001',
+            'function (doc) { index("default", doc._id); }}'
+            )
+        self.assertEqual(search.design_doc, ddoc)
+        self.assertEqual(search.index_name, 'searchindex001')
+        self.assertIsInstance(search['index'], _Code)
+        self.assertEqual(
+            search['index'],
+            'function (doc) { index("default", doc._id); }}'
+        )
+        self.assertIsInstance(search['analyzer'], _Code)
+        self.assertEqual(
+            search['analyzer'],
+            'standard'
+        )
+
+        self.assertEqual(search, {
+            'index': 'function (doc) { index("default", doc._id); }}',
+            'analyzer': 'standard'
+        })
```

Comment:
You have a few branches in your setter, is it worth testing the `None` cases for the name and func args?